frontend/tests/api/user/saved-opportunities/route.test.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @jest-environment node
+ */
+
+import { DELETE, POST } from "src/app/api/user/saved-opportunities/route";
+
+import { NextRequest } from "next/server";
+
+const getSessionMock = jest.fn();
+
+jest.mock("src/services/auth/session", () => ({
+  getSession: (): unknown => getSessionMock(),
+}));
+
+const mockPostSavedOpp = jest.fn((params: unknown): unknown =>
+  Promise.resolve({ json: () => Promise.resolve(params) }),
+);
+
+jest.mock("src/services/fetch/fetchers/savedOpportunityFetcher", () => ({
+  handleSavedOpportunity: () => mockPostSavedOpp({ status_code: 200 }),
+}));
+
+const fakeRequestForSavedOpps = () => {
+  return {
+    headers: {
+      get: jest.fn(() => {
+        return {
+          opportunity_id: 1,
+        };
+      }),
+    },
+  } as unknown as NextRequest;
+};
+
+describe("POST and DELETE request", () => {
```

Comment:
similar to the other test, can you add a couple of error tests?

Reply:
Added, thanks.